packages/next/client/page-loader.js

```diff
@@ -177,19 +183,21 @@ export default class PageLoader {
   prefetchData(href, asPath) {
     const { pathname: hrefPathname } = parse(href, true)
     const route = normalizeRoute(hrefPathname)
-    return this.promisedSsgManifest.then(
-      (s, _dataHref) =>
-        // Check if the route requires a data file
-        s.has(route) &&
-        // Try to generate data href, noop when falsy
-        (_dataHref = this.getDataHref(href, asPath)) &&
-        // noop when data has already been prefetched (dedupe)
-        !document.querySelector(
-          `link[rel="${relPrefetch}"][href^="${_dataHref}"]`
-        ) &&
-        // Inject the `<link rel=prefetch>` tag for above computed `href`.
-        appendLink(_dataHref, relPrefetch, 'fetch')
-    )
+    requestIdleCallback(() => {
```

Comment:
This interrupted the promise return flow. Can we be sure this is promisified and still return please?